web3_proxy/src/bin/web3_proxy_cli/proxyd.rs

```diff
@@ -143,70 +154,160 @@ async fn run(
         }
     }
 
+    // start thread for watching config
+    // if let Some(top_config_path) = top_config_path {
+    //     let config_sender = spawned_app.new_top_config_sender;
+    //     {
+    //         thread::spawn(move || loop {
+    //             match fs::read_to_string(&top_config_path) {
+    //                 Ok(new_top_config) => match toml::from_str(&new_top_config) {
+    //                     Ok(new_top_config) => {
+    //                         if new_top_config != top_config {
+    //                             top_config = new_top_config;
+    //                             config_sender.send(top_config.clone()).unwrap();
+    //                         }
+    //                     }
+    //                     Err(err) => {
+    //                         // TODO: panic?
+    //                         error!("Unable to parse config! {:#?}", err);
+    //                     }
+    //                 },
+    //                 Err(err) => {
+    //                     // TODO: panic?
+    //                     error!("Unable to read config! {:#?}", err);
+    //                 }
+    //             }
+    //
+    //             thread::sleep(Duration::from_secs(10));
+    //         });
+    //     }
+    // }
+
     // start the prometheus metrics port
-    let prometheus_handle = tokio::spawn(metrics_frontend::serve(
+    let prometheus_handle = tokio::spawn(prometheus::serve(
         spawned_app.app.clone(),
         app_prometheus_port,
+        prometheus_shutdown_receiver,
     ));
 
     // wait until the app has seen its first consensus head block
-    // TODO: if backups were included, wait a little longer?
+    // if backups were included, wait a little longer
+    // let _ = spawned_app.app.head_block_receiver().changed().await;
+    // if backups were included, wait a little longer
+    // for _ in 0..3 {
+    //     let _ = spawned_app.consensus_connections_watcher.changed().await;
+    //
+    //     let consensus = spawned_app
+    //         .consensus_connections_watcher
+    //         .borrow_and_update();
+    //
+    //     // Let's just do a super dirty unwrap to get things going
+    //     if consensus.unwrap().backups_needed {
+    //         info!(
+    //             "waiting longer. found consensus with backups: {}",
+    //             consensus.unwrap().head_block.as_ref().unwrap(),
+    //         );
+    //     } else {
+    //         // TODO: also check that we have at least one archive node connected?
+    //         break;
+    //     }
+    // }
```

Comment:
why is this commented out?

Reply:
again, I think the commented out version was replaced by the line below https://github.com/yenicelik/web3-proxy/blob/stats_v3/web3_proxy/src/bin/web3_proxy_cli/proxyd.rs#L193 

let me know if this is the case or not